benchmark/manage_gatling_client.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+set -e
+set -o pipefail
+# set -x
+#Functions
+function usage {
+  echo "Setup of test clients for the benchmark test with a pre-defined secret"
+  echo "Usage: $(basename $0) -k KEYCLOAK_HOME -r REALM_NAME -c CLIENT_ID -s SERVER_URL [-d]" 2>&1
+  echo '-k keycloak home path ex: /home/opt/keycloak-18.0.0 : default value KEYCLOAK_HOME env variable'
+  echo '-r realm : default value realm-0'
+  echo '-c service account enabled client name : default value gatling'
+  echo '-s keycloak host server url : default value http://localhost:8081/auth'
+  echo '-d delete the client and realm before re-creating them: default value is false'
+  exit 1
+}
+
+function install_kcb {
+  echo "Installing kcadm.sh"
+  export PATH=$PATH:$KEYCLOAK_HOME/bin
+}
+
+function create_realm {
+  if kcadm.sh get realms/${REALM_NAME}; then
+    echo "INFO: Skipping Realm creation as realm exists"
+  else
+    echo "INFO: Creating Realm with realm id: ${REALM_NAME}"
+    kcadm.sh create realms -s realm=$REALM_NAME -s enabled=true -o
+  fi
+}
+
+function create_client_assign_roles {
+  #Create the client application with Service Account Roles setup
+  CID=$(kcadm.sh create clients -r $REALM_NAME -s clientId=$CLIENT_ID -s enabled=true -i)
+  echo "INFO: Created New ${CLIENT_ID} Client with Client ID ${CID}"
+  #Update the client with necessary attributes
+  kcadm.sh update clients/$CID -r $REALM_NAME  -s "secret=setup-for-benchmark" -s serviceAccountsEnabled=true  -s publicClient=false -s 'redirectUris=["http://localhost:8081"]'
+  kcadm.sh get clients/$CID -r $REALM_NAME --fields clientId,id,redirectUris,publicClient,serviceAccountsEnabled,secret
+
+  #Verify if the service account user exists
+  kcadm.sh get users -r $REALM_NAME -q username=service-account-$CLIENT_ID --fields id,username,enabled
+
```

Comment:
It seems to me that the two "kcadm.sh get" commands are a left-over debugging the script and getting the initial functionality in place. Please consider removing.

Reply:
yep, they are not needed in the current state of the script. will remove them